packages/jest-jasmine2/src/index.js

```diff
@@ -39,6 +39,28 @@ function jasmine2(config, environment, moduleLoader, testPath) {
 
     const requireJasmine = environment.global.jasmineRequire;
     jasmine = requireJasmine.core(requireJasmine);
+
+    const jasmineBuildExpectationResult = jasmine.buildExpectationResult;
+
+    // @see https://github.com/facebook/jest/issues/429
+    //
+    // Because we build error messages after the test is complete we can run
+    // into reporting issues.
+    // @example
+    //    let a = [1];
+    //    expect(a).toEqual([1, 2]);
+    //    a.push(2); // mutate the original value after it fails the matcher
+    // will print `expected [2] to equal [2]`
+    //
+    // The workaround is to clone the values at the time of the failure. That
+    // way we hold the value that won't be mutated during the test execution.
+    jasmine.buildExpectationResult = function(options) {
+      options.expected = JSON.parse(JSON.stringify(options.expected));
+      options.actual = JSON.parse(JSON.stringify(options.actual));
```

Comment:
yeah this isn't great. I think shallow copy would definitely be better? Otherwise Jasmine might have a utility that we could use?

Also, let's remove all those comments. They should be part of the commit message and can then be found when blaming these lines of code :) Don't need all this explanation in there.